# Processing Social Habituation Task Scans

## 1. Examine raw data for problems.

In [10]:
cd /ifs/faculty/mirella/fmri/Prisma/
fslview_deprecated

fslview_bin: cannot connect to X server localhost:341.0


: 1

In [1]:
source "$GUNTHERDIR/include/Asourceall.sh"

## 2. Add new subjects
### Subjects_all, subjects_new, subject_[pres_order], and  subject_[pres_order]_new


In [23]:
subjects_all="TD182B TD174B ASD180B ASD189B ASD156B ASD148D ASD113F TD157B ASD158B ASD187B TD106D TD163B ASD200B TD166B ASD194B ASD163B ASD162B TD170B ASD167B ASD159B ASD192B"
subjects_new="TD170B" # "ASD167B ASD159B ASD192B" 

subjects_a="TD182B ASD180B ASD113F TD163B ASD194B ASD159B "
subjects_a_new="ASD159B"
subjects_b="TD174B ASD189B ASD158B ASD163B ASD192B TD170B "
subjects_b_new="TD170B" #ASD192B
subjects_c="ASD156B TD157B ASD187B TD166B TD162B "
subjects_c_new="ASD162B"
subjects_d="ASD148D TD106D ASD200B ASD167B "
subjects_d_new="ASD167B"


## 3. Register fMRI to MPRAGE

In [3]:
start=$(date +%s.%N)

$GUNTHERDIR/g_regFunc.sh -B /ifs/faculty/mirella/fmri/Prisma -A mprage_HCP4min -R socialhabituation -S ${subjects_new}

end=$(date +%s.%N)    
runtime=$(python -c "print(${end} - ${start})")

echo "Runtime was $runtime"

Executing: '/ifs/faculty/mirella/neurogunther/g_regFunc.sh -B /ifs/faculty/mirella/fmri/Prisma -A mprage_HCP4min -R socialhabituation -S ASD167B ASD159B ASD192B TD170B'
reading list of subjects given by user
Running subjects: ASD167B ASD159B ASD192B TD170B


Running ASD167B
X: cd /ifs/faculty/mirella/fmri/Prisma/subjects/ASD167B/socialhabituation
X: mkdir /ifs/faculty/mirella/fmri/Prisma/subjects/ASD167B/socialhabituation/2_reg 2> /dev/null; echo -n ''
X: cd /ifs/faculty/mirella/fmri/Prisma/subjects/ASD167B/socialhabituation/2_reg
/ifs/faculty/mirella/neurogunther/include/zogger.sh: line 53: cd: /ifs/faculty/mirella/fmri/Prisma/subjects/ASD167B/socialhabituation/2_reg: No such file or directory
*ERROR* non-zero exit while executing 'cd /ifs/faculty/mirella/f.....'

Running ASD159B
*ERROR* Input file '/ifs/faculty/mirella/fmri/Prisma/subjects/ASD159B/mprage_HCP4min/brain.nii.gz' does not exist
skipping subject ASD159B
/ifs/faculty/mirella/neurogunther/include/loops.sh: line 79: return: 

____
<span style="font-family:courier;">Runtime for 1 subject on BMAP was 382 sec = 6.3 min</span>

## 4. Make Registration and Motion Reports
### 4. A. Registration Reports
#### Make reports

In [5]:
cd /ifs/faculty/mirella/fmri/Prisma/ShulaK08/SocialHabituation/reports/registration
gh_reportReg.sh report_reg_ASD_SocialHabituation.html ../../../../subjects/ASD*/socialhabituation/2_reg
gh_reportReg.sh report_reg_TD_SocialHabituation.html ../../../../subjects/TD*/socialhabituation/2_reg

#### View reports
Copy/paste into browser on BMAP:
<file:///ifs/faculty/mirella/fmri/Prisma/ShulaK08/SocialHabituation/reports/registration/report_reg_ASD_SocialHabituation.html>

Copy/paste into browser on your computer if using OSXFUSE or similar:
<file:///Users/emilywood/Desktop/BMAP/Prisma/ShulaK08/SocialHabituation/reports/registration/report_reg_TD_SocialHabituation.html>

### 4. B. Motion Reports
#### Make reports

In [6]:
cd /ifs/faculty/mirella/fmri/Prisma/ShulaK08/SocialHabituation/reports/motion
gh_reportMotion.sh report_motion_ASD_sochabit.html ../../../../subjects/ASD*/socialhabituation/1_preprocess/mc
gh_reportMotion.sh report_motion_TD_sochabit.html ../../../../subjects/TD*/socialhabituation/1_preprocess/mc


Fatal Signal 13 (SIGPIPE) received
    mri_write_ascii
   mri_write_1D
  1dTsort main
 Bottom of Debug Stack
** AFNI version = AFNI_16.0.00  Compile date = Jan  1 2016
** [[Precompiled binary linux_xorg7_64: Jan  1 2016]]
** Program Death **
** If you report this crash to the AFNI message board,
** please copy the error messages EXACTLY, and give
** the command line you used to run the program, and
** any other information needed to repeat the problem.
** You may later be asked to upload data to help debug.
** Crash log is appended to file /ifshome/ewood/.afni.crashlog

Fatal Signal 13 (SIGPIPE) received
    mri_write_ascii
   mri_write_1D
  1dTsort main
 Bottom of Debug Stack
** AFNI version = AFNI_16.0.00  Compile date = Jan  1 2016
** [[Precompiled binary linux_xorg7_64: Jan  1 2016]]
** Program Death **
** If you report this crash to the AFNI message board,
** please copy the error messages EXACTLY, and give
** the command line you used to run the program, and
** any other informa

** [[Precompiled binary linux_xorg7_64: Jan  1 2016]]
** Program Death **
** If you report this crash to the AFNI message board,
** please copy the error messages EXACTLY, and give
** the command line you used to run the program, and
** any other information needed to repeat the problem.
** You may later be asked to upload data to help debug.
** Crash log is appended to file /ifshome/ewood/.afni.crashlog

Fatal Signal 13 (SIGPIPE) received
    mri_write_ascii
   mri_write_1D
  1dTsort main
 Bottom of Debug Stack
** AFNI version = AFNI_16.0.00  Compile date = Jan  1 2016
** [[Precompiled binary linux_xorg7_64: Jan  1 2016]]
** Program Death **
** If you report this crash to the AFNI message board,
** please copy the error messages EXACTLY, and give
** the command line you used to run the program, and
** any other information needed to repeat the problem.
** You may later be asked to upload data to help debug.
** Crash log is appended to file /ifshome/ewood/.afni.crashlog

Fatal Signal 1

** please copy the error messages EXACTLY, and give
** the command line you used to run the program, and
** any other information needed to repeat the problem.
** You may later be asked to upload data to help debug.
** Crash log is appended to file /ifshome/ewood/.afni.crashlog

Fatal Signal 13 (SIGPIPE) received
    mri_write_ascii
   mri_write_1D
  1dTsort main
 Bottom of Debug Stack
** AFNI version = AFNI_16.0.00  Compile date = Jan  1 2016
** [[Precompiled binary linux_xorg7_64: Jan  1 2016]]
** Program Death **
** If you report this crash to the AFNI message board,
** please copy the error messages EXACTLY, and give
** the command line you used to run the program, and
** any other information needed to repeat the problem.
** You may later be asked to upload data to help debug.
** Crash log is appended to file /ifshome/ewood/.afni.crashlog

Fatal Signal 13 (SIGPIPE) received
    mri_write_ascii
   mri_write_1D
  1dTsort main
 Bottom of Debug Stack
** AFNI version = AFNI_16.0.00  Co

** You may later be asked to upload data to help debug.
** Crash log is appended to file /ifshome/ewood/.afni.crashlog

Fatal Signal 13 (SIGPIPE) received
    mri_write_ascii
   mri_write_1D
  1dTsort main
 Bottom of Debug Stack
** AFNI version = AFNI_16.0.00  Compile date = Jan  1 2016
** [[Precompiled binary linux_xorg7_64: Jan  1 2016]]
** Program Death **
** If you report this crash to the AFNI message board,
** please copy the error messages EXACTLY, and give
** the command line you used to run the program, and
** any other information needed to repeat the problem.
** You may later be asked to upload data to help debug.
** Crash log is appended to file /ifshome/ewood/.afni.crashlog

Fatal Signal 13 (SIGPIPE) received
    mri_write_ascii
   mri_write_1D
  1dTsort main
 Bottom of Debug Stack
** AFNI version = AFNI_16.0.00  Compile date = Jan  1 2016
** [[Precompiled binary linux_xorg7_64: Jan  1 2016]]
** Program Death **
** If you report this crash to the AFNI message board,
** ple


Fatal Signal 13 (SIGPIPE) received
    mri_write_ascii
   mri_write_1D
  1dTsort main
 Bottom of Debug Stack
** AFNI version = AFNI_16.0.00  Compile date = Jan  1 2016
** [[Precompiled binary linux_xorg7_64: Jan  1 2016]]
** Program Death **
** If you report this crash to the AFNI message board,
** please copy the error messages EXACTLY, and give
** the command line you used to run the program, and
** any other information needed to repeat the problem.
** You may later be asked to upload data to help debug.
** Crash log is appended to file /ifshome/ewood/.afni.crashlog

Fatal Signal 13 (SIGPIPE) received
    mri_write_ascii
   mri_write_1D
  1dTsort main
 Bottom of Debug Stack
** AFNI version = AFNI_16.0.00  Compile date = Jan  1 2016
** [[Precompiled binary linux_xorg7_64: Jan  1 2016]]
** Program Death **
** If you report this crash to the AFNI message board,
** please copy the error messages EXACTLY, and give
** the command line you used to run the program, and
** any other informa

** [[Precompiled binary linux_xorg7_64: Jan  1 2016]]
** Program Death **
** If you report this crash to the AFNI message board,
** please copy the error messages EXACTLY, and give
** the command line you used to run the program, and
** any other information needed to repeat the problem.
** You may later be asked to upload data to help debug.
** Crash log is appended to file /ifshome/ewood/.afni.crashlog

Fatal Signal 13 (SIGPIPE) received
    mri_write_ascii
   mri_write_1D
  1dTsort main
 Bottom of Debug Stack
** AFNI version = AFNI_16.0.00  Compile date = Jan  1 2016
** [[Precompiled binary linux_xorg7_64: Jan  1 2016]]
** Program Death **
** If you report this crash to the AFNI message board,
** please copy the error messages EXACTLY, and give
** the command line you used to run the program, and
** any other information needed to repeat the problem.
** You may later be asked to upload data to help debug.
** Crash log is appended to file /ifshome/ewood/.afni.crashlog

Fatal Signal 1

#### View Reports
Copy/paste into browser on BMAP: <file:///ifs/faculty/mirella/fmri/Prisma/ShulaK08/SocialHabituation/reports/motion/report_motion_ASD_SocialHabituation.html>
Copy/paste into browser on your computer if using OSXFUSE or similar: <file:///Users/emilywood/Desktop/BMAP/Prisma/ShulaK08/SocialHabituation/reports/motion/report_motion_TD_SocialHabituation.html>

#### Record motion parameters in Spreadsheet 
(SocialHabituation_Data_date_.xls in BMAP folder)
Evaluate registration and note in Spreadsheet



## 5. Put Habituation EVs in each Subject's folder
### Chose the correct presentation order (A, B, C, or D)

In [8]:
for subject in ${subjects_a_new}; do
zcmd "mkdir /ifs/faculty/mirella/fmri/Prisma/subjects/${subject}/socialhabituation/EVs"
zcmd "cp -r /ifs/faculty/mirella/fmri/Prisma/ShulaK08/SocialHabituation/EVs/JointNonSocHabitA.txt /ifs/faculty/mirella/fmri/Prisma/subjects/${subject}/socialhabituation/EVs/"
zcmd "cp -r /ifs/faculty/mirella/fmri/Prisma/ShulaK08/SocialHabituation/EVs/JointSocHabitA.txt /ifs/faculty/mirella/fmri/Prisma/subjects/${subject}/socialhabituation/EVs/"
zcmd "cp -r /ifs/faculty/mirella/fmri/Prisma/ShulaK08/SocialHabituation/EVs/AudNonSocHabitA.txt /ifs/faculty/mirella/fmri/Prisma/subjects/${subject}/socialhabituation/EVs/"
zcmd "cp -r /ifs/faculty/mirella/fmri/Prisma/ShulaK08/SocialHabituation/EVs/AudSocHabitA.txt /ifs/faculty/mirella/fmri/Prisma/subjects/${subject}/socialhabituation/EVs/"
zcmd "cp -r /ifs/faculty/mirella/fmri/Prisma/ShulaK08/SocialHabituation/EVs/TactSocHabitAC.txt /ifs/faculty/mirella/fmri/Prisma/subjects/${subject}/socialhabituation/EVs/"
zcmd "mv /ifs/faculty/mirella/fmri/Prisma/subjects/${subject}/socialhabituation/EVs/JointNonSocHabitA.txt /ifs/faculty/mirella/fmri/Prisma/subjects/${subject}/socialhabituation/EVs/JointNonSocHabit.txt"
zcmd "mv /ifs/faculty/mirella/fmri/Prisma/subjects/${subject}/socialhabituation/EVs/JointSocHabitA.txt /ifs/faculty/mirella/fmri/Prisma/subjects/${subject}/socialhabituation/EVs/JointSocHabit.txt"
zcmd "mv /ifs/faculty/mirella/fmri/Prisma/subjects/${subject}/socialhabituation/EVs/AudNonSocHabitA.txt /ifs/faculty/mirella/fmri/Prisma/subjects/${subject}/socialhabituation/EVs/AudNonSocHabit.txt"
zcmd "mv /ifs/faculty/mirella/fmri/Prisma/subjects/${subject}/socialhabituation/EVs/AudSocHabitA.txt /ifs/faculty/mirella/fmri/Prisma/subjects/${subject}/socialhabituation/EVs/AudSocHabit.txt"
zcmd "mv /ifs/faculty/mirella/fmri/Prisma/subjects/${subject}/socialhabituation/EVs/TactSocHabitAC.txt /ifs/faculty/mirella/fmri/Prisma/subjects/${subject}/socialhabituation/EVs/TactSocHabit.txt"
done

for subject in ${subjects_b_new}; do
zcmd "mkdir /ifs/faculty/mirella/fmri/Prisma/subjects/${subject}/socialhabituation/EVs"
zcmd "cp -r /ifs/faculty/mirella/fmri/Prisma/ShulaK08/SocialHabituation/EVs/JointNonSocHabitB.txt /ifs/faculty/mirella/fmri/Prisma/subjects/${subject}/socialhabituation/EVs/"
zcmd "cp -r /ifs/faculty/mirella/fmri/Prisma/ShulaK08/SocialHabituation/EVs/JointSocHabitB.txt /ifs/faculty/mirella/fmri/Prisma/subjects/${subject}/socialhabituation/EVs/"
zcmd "cp -r /ifs/faculty/mirella/fmri/Prisma/ShulaK08/SocialHabituation/EVs/AudNonSocHabitB.txt /ifs/faculty/mirella/fmri/Prisma/subjects/${subject}/socialhabituation/EVs/"
zcmd "cp -r /ifs/faculty/mirella/fmri/Prisma/ShulaK08/SocialHabituation/EVs/AudSocHabitB.txt /ifs/faculty/mirella/fmri/Prisma/subjects/${subject}/socialhabituation/EVs/"
zcmd "cp -r /ifs/faculty/mirella/fmri/Prisma/ShulaK08/SocialHabituation/EVs/TactSocHabitBD.txt /ifs/faculty/mirella/fmri/Prisma/subjects/${subject}/socialhabituation/EVs/"
zcmd "mv /ifs/faculty/mirella/fmri/Prisma/subjects/${subject}/socialhabituation/EVs/JointNonSocHabitB.txt /ifs/faculty/mirella/fmri/Prisma/subjects/${subject}/socialhabituation/EVs/JointNonSocHabit.txt"
zcmd "mv /ifs/faculty/mirella/fmri/Prisma/subjects/${subject}/socialhabituation/EVs/JointSocHabitB.txt /ifs/faculty/mirella/fmri/Prisma/subjects/${subject}/socialhabituation/EVs/JointSocHabit.txt"
zcmd "mv /ifs/faculty/mirella/fmri/Prisma/subjects/${subject}/socialhabituation/EVs/AudNonSocHabitB.txt /ifs/faculty/mirella/fmri/Prisma/subjects/${subject}/socialhabituation/EVs/AudNonSocHabit.txt"
zcmd "mv /ifs/faculty/mirella/fmri/Prisma/subjects/${subject}/socialhabituation/EVs/AudSocHabitB.txt /ifs/faculty/mirella/fmri/Prisma/subjects/${subject}/socialhabituation/EVs/AudSocHabit.txt"
zcmd "mv /ifs/faculty/mirella/fmri/Prisma/subjects/${subject}/socialhabituation/EVs/TactSocHabitBD.txt /ifs/faculty/mirella/fmri/Prisma/subjects/${subject}/socialhabituation/EVs/TactSocHabit.txt"
done

for subject in ${subjects_c_new}; do
zcmd "mkdir /ifs/faculty/mirella/fmri/Prisma/subjects/${subject}/socialhabituation/EVs"
zcmd "cp -r /ifs/faculty/mirella/fmri/Prisma/ShulaK08/SocialHabituation/EVs/JointNonSocHabitC.txt /ifs/faculty/mirella/fmri/Prisma/subjects/${subject}/socialhabituation/EVs/"
zcmd "cp -r /ifs/faculty/mirella/fmri/Prisma/ShulaK08/SocialHabituation/EVs/JointSocHabitC.txt /ifs/faculty/mirella/fmri/Prisma/subjects/${subject}/socialhabituation/EVs/"
zcmd "cp -r /ifs/faculty/mirella/fmri/Prisma/ShulaK08/SocialHabituation/EVs/AudNonSocHabitC.txt /ifs/faculty/mirella/fmri/Prisma/subjects/${subject}/socialhabituation/EVs/"
zcmd "cp -r /ifs/faculty/mirella/fmri/Prisma/ShulaK08/SocialHabituation/EVs/AudSocHabitC.txt /ifs/faculty/mirella/fmri/Prisma/subjects/${subject}/socialhabituation/EVs/"
zcmd "cp -r /ifs/faculty/mirella/fmri/Prisma/ShulaK08/SocialHabituation/EVs/TactSocHabitAC.txt /ifs/faculty/mirella/fmri/Prisma/subjects/${subject}/socialhabituation/EVs/"
zcmd "mv /ifs/faculty/mirella/fmri/Prisma/subjects/${subject}/socialhabituation/EVs/JointNonSocHabitC.txt /ifs/faculty/mirella/fmri/Prisma/subjects/${subject}/socialhabituation/EVs/JointNonSocHabit.txt"
zcmd "mv /ifs/faculty/mirella/fmri/Prisma/subjects/${subject}/socialhabituation/EVs/JointSocHabitC.txt /ifs/faculty/mirella/fmri/Prisma/subjects/${subject}/socialhabituation/EVs/JointSocHabit.txt"
zcmd "mv /ifs/faculty/mirella/fmri/Prisma/subjects/${subject}/socialhabituation/EVs/AudNonSocHabitC.txt /ifs/faculty/mirella/fmri/Prisma/subjects/${subject}/socialhabituation/EVs/AudNonSocHabit.txt"
zcmd "mv /ifs/faculty/mirella/fmri/Prisma/subjects/${subject}/socialhabituation/EVs/AudSocHabitC.txt /ifs/faculty/mirella/fmri/Prisma/subjects/${subject}/socialhabituation/EVs/AudSocHabit.txt"
zcmd "mv /ifs/faculty/mirella/fmri/Prisma/subjects/${subject}/socialhabituation/EVs/TactSocHabitAC.txt /ifs/faculty/mirella/fmri/Prisma/subjects/${subject}/socialhabituation/EVs/TactSocHabit.txt"
done

for subject in ${subjects_d_new}; do
zcmd "mkdir /ifs/faculty/mirella/fmri/Prisma/subjects/${subject}/socialhabituation/EVs"
zcmd "cp -r /ifs/faculty/mirella/fmri/Prisma/ShulaK08/SocialHabituation/EVs/JointNonSocHabitD.txt /ifs/faculty/mirella/fmri/Prisma/subjects/${subject}/socialhabituation/EVs/"
zcmd "cp -r /ifs/faculty/mirella/fmri/Prisma/ShulaK08/SocialHabituation/EVs/JointSocHabitD.txt /ifs/faculty/mirella/fmri/Prisma/subjects/${subject}/socialhabituation/EVs/"
zcmd "cp -r /ifs/faculty/mirella/fmri/Prisma/ShulaK08/SocialHabituation/EVs/AudNonSocHabitD.txt /ifs/faculty/mirella/fmri/Prisma/subjects/${subject}/socialhabituation/EVs/"
zcmd "cp -r /ifs/faculty/mirella/fmri/Prisma/ShulaK08/SocialHabituation/EVs/AudSocHabitD.txt /ifs/faculty/mirella/fmri/Prisma/subjects/${subject}/socialhabituation/EVs/"
zcmd "cp -r /ifs/faculty/mirella/fmri/Prisma/ShulaK08/SocialHabituation/EVs/TactSocHabitBD.txt /ifs/faculty/mirella/fmri/Prisma/subjects/${subject}/socialhabituation/EVs/"
zcmd "mv /ifs/faculty/mirella/fmri/Prisma/subjects/${subject}/socialhabituation/EVs/JointNonSocHabitD.txt /ifs/faculty/mirella/fmri/Prisma/subjects/${subject}/socialhabituation/EVs/JointNonSocHabit.txt"
zcmd "mv /ifs/faculty/mirella/fmri/Prisma/subjects/${subject}/socialhabituation/EVs/JointSocHabitD.txt /ifs/faculty/mirella/fmri/Prisma/subjects/${subject}/socialhabituation/EVs/JointSocHabit.txt"
zcmd "mv /ifs/faculty/mirella/fmri/Prisma/subjects/${subject}/socialhabituation/EVs/AudNonSocHabitD.txt /ifs/faculty/mirella/fmri/Prisma/subjects/${subject}/socialhabituation/EVs/AudNonSocHabit.txt"
zcmd "mv /ifs/faculty/mirella/fmri/Prisma/subjects/${subject}/socialhabituation/EVs/AudSocHabitD.txt /ifs/faculty/mirella/fmri/Prisma/subjects/${subject}/socialhabituation/EVs/AudSocHabit.txt"
zcmd "mv /ifs/faculty/mirella/fmri/Prisma/subjects/${subject}/socialhabituation/EVs/TactSocHabitBD.txt /ifs/faculty/mirella/fmri/Prisma/subjects/${subject}/socialhabituation/EVs/TactSocHabit.txt"
done

X: mkdir /ifs/faculty/mirella/fmri/Prisma/subjects/TD170B/socialhabituation/EVs
X: cp -r /ifs/faculty/mirella/fmri/Prisma/ShulaK08/SocialHabituation/EVs/JointNonSocHabitB.txt /ifs/faculty/mirella/fmri/Prisma/subjects/TD170B/socialhabituation/EVs/
X: cp -r /ifs/faculty/mirella/fmri/Prisma/ShulaK08/SocialHabituation/EVs/JointSocHabitB.txt /ifs/faculty/mirella/fmri/Prisma/subjects/TD170B/socialhabituation/EVs/
X: cp -r /ifs/faculty/mirella/fmri/Prisma/ShulaK08/SocialHabituation/EVs/AudNonSocHabitB.txt /ifs/faculty/mirella/fmri/Prisma/subjects/TD170B/socialhabituation/EVs/
X: cp -r /ifs/faculty/mirella/fmri/Prisma/ShulaK08/SocialHabituation/EVs/AudSocHabitB.txt /ifs/faculty/mirella/fmri/Prisma/subjects/TD170B/socialhabituation/EVs/
X: cp -r /ifs/faculty/mirella/fmri/Prisma/ShulaK08/SocialHabituation/EVs/TactSocHabitBD.txt /ifs/faculty/mirella/fmri/Prisma/subjects/TD170B/socialhabituation/EVs/
X: mv /ifs/faculty/mirella/fmri/Prisma/subjects/TD170B/socialhabituation/EVs/JointNonSocHabitB.txt

## 6. Run FSL motion outliers script


In [10]:
start=$(date +%s.%N)

baseDir="/ifs/faculty/mirella/fmri/Prisma/subjects" #subjects folder; no slash at end
inputDir="socialhabituation" 
outputDir="/ifs/faculty/mirella/fmri/Prisma/subjects/socialhabituation" 

for subject in ${subjects_new}; do
zcmd "cd ${baseDir}/${subject}/${inputDir}"
zcmd "fsl_motion_outliers -i func_task_pp.nii.gz --refrms --nomoco -p refrms_plot -o motion_scrub_refrms"
done

end=$(date +%s.%N)    
runtime=$(python -c "print(${end} - ${start})")

echo "Runtime was $runtime"

X: cd /ifs/faculty/mirella/fmri/Prisma/subjects/TD170B/socialhabituation
X: fsl_motion_outliers -i func_task_pp.nii.gz --refrms --nomoco -p refrms_plot -o motion_scrub_refrms
Runtime was 344.2520503997803


____
<span style="font-family:courier;">Runtime for 1 subject on BMAP was 344 sec = 5.75 min</span>

## 7. Run FEATs


In [12]:
start=$(date +%s.%N)

zcmd "$GUNTHERDIR/g_runFeats.sh -T /ifs/faculty/mirella/fmri/Prisma/ShulaK08/SocialHabituation/design/design.fsf -j K08PILOT_121917 -B /ifs/faculty/mirella/fmri/Prisma -R socialhabituation -S $subjects_new"
zcmd "$GUNTHERDIR/g_runFeats.sh -T /ifs/faculty/mirella/fmri/Prisma/ShulaK08/SocialHabituation/design/designmp_motionscrub.fsf -j K08PILOT_121917 -B /ifs/faculty/mirella/fmri/Prisma -R socialhabituation -S $subjects_new"

end=$(date +%s.%N)    
runtime=$(python -c "print(${end} - ${start})")

echo "Runtime was $runtime"

X: /ifs/faculty/mirella/neurogunther/g_runFeats.sh -T /ifs/faculty/mirella/fmri/Prisma/ShulaK08/SocialHabituation/design/design.fsf -j K08PILOT_121917 -B /ifs/faculty/mirella/fmri/Prisma -R socialhabituation -S TD170B
Executing: '/ifs/faculty/mirella/neurogunther/g_runFeats.sh -T /ifs/faculty/mirella/fmri/Prisma/ShulaK08/SocialHabituation/design/design.fsf -j K08PILOT_121917 -B /ifs/faculty/mirella/fmri/Prisma -R socialhabituation -S TD170B'
reading list of subjects given by user
Running subjects: TD170B


Running TD170B
...skipping check of final output
X: mkdir -p /ifs/faculty/mirella/fmri/Prisma/subjects/TD170B/socialhabituation/feats/fsf 2> /dev/null; echo -n ''
X: /ifs/faculty/mirella/neurogunther/x_fsfsubstitute.sh -i /ifs/faculty/mirella/fmri/Prisma/ShulaK08/SocialHabituation/design/design.fsf -f K08PILOT_121917 -r TD170B -o /ifs/faculty/mirella/fmri/Prisma/subjects/TD170B/socialhabituation/feats/fsf/design.fsf -c '1 0 0'
Creating fsf file
X: sed s/K08PILOT_121917/TD170B/g /ifs/

absolute: /ifs/faculty/mirella/fmri/Prisma/subjects/TD170B/socialhabituation/feats/simplemodelmp_motionscrub.feat/reg -> /ifs/faculty/mirella/fmri/Prisma/subjects/TD170B/socialhabituation/2_reg
changed:  /ifs/faculty/mirella/fmri/Prisma/subjects/TD170B/socialhabituation/feats/simplemodelmp_motionscrub.feat/reg -> ../../2_reg
absolute: /ifs/faculty/mirella/fmri/Prisma/subjects/TD170B/socialhabituation/feats/simplemodelmp_motionscrub.feat/mc -> /ifs/faculty/mirella/fmri/Prisma/subjects/TD170B/socialhabituation/1_preprocess/mc
changed:  /ifs/faculty/mirella/fmri/Prisma/subjects/TD170B/socialhabituation/feats/simplemodelmp_motionscrub.feat/mc -> ../../1_preprocess/mc


You did not have any errors/warnings...great job good-looking. Time for a beer!
Runtime was 28.885639429092407


____
Even though it says you were done in 29 sec, the FEAT is running in the background.

<span style="font-family:courier;">Runtime for 1 subject on BMAP was X sec = X min</span>